File: ruoyi-ui/src/constants/reportState.js
/**
 * 实验报告状态常量
 * 对应后端 ReportState 枚举
 */
export const ReportState = {
  DRAFT: '0',           // 草稿
  SUBMITTED: '1',       // 已提交
  REVIEWING: '2',       // 批阅中
  REVIEWED: '3',        // 已批阅
  REJECTED: '4',        // 已打回
  RESUBMITTED: '5',     // 重新提交
  ARCHIVED: '6'         // 已归档
}

/**
 * 状态描述映射
 */
export const ReportStateDesc = {
  [ReportState.DRAFT]: '草稿',
  [ReportState.SUBMITTED]: '已提交',
  [ReportState.REVIEWING]: '批阅中',
  [ReportState.REVIEWED]: '已批阅',
  [ReportState.REJECTED]: '已打回',
  [ReportState.RESUBMITTED]: '重新提交',
  [ReportState.ARCHIVED]: '已归档'
}

/**
 * 状态标签类型映射（Element UI Tag type）
 */
export const ReportStateType = {
  [ReportState.DRAFT]: 'info',
  [ReportState.SUBMITTED]: 'primary',
  [ReportState.REVIEWING]: 'warning',
  [ReportState.REVIEWED]: 'success',
  [ReportState.REJECTED]: 'danger',
  [ReportState.RESUBMITTED]: 'primary',
  [ReportState.ARCHIVED]: 'info'
}

/**
 * 触发器常量
 * 对应后端 ReportTrigger 枚举
 */
export const ReportTrigger = {
  SUBMIT: 'SUBMIT',               // 提交报告
  START_REVIEW: 'START_REVIEW',   // 开始批阅
  APPROVE: 'APPROVE',             // 批阅通过
  REJECT: 'REJECT',               // 打回
  RESUBMIT: 'RESUBMIT',           // 重新提交
  ARCHIVE: 'ARCHIVE'              // 归档
}

/**
 * 触发器描述映射
 */
export const ReportTriggerDesc = {
  [ReportTrigger.SUBMIT]: '提交',
  [ReportTrigger.START_REVIEW]: '开始批阅',
  [ReportTrigger.APPROVE]: '批阅通过',
  [ReportTrigger.REJECT]: '打回',
  [ReportTrigger.RESUBMIT]: '重新提交',
  [ReportTrigger.ARCHIVE]: '归档'
}

/**
 * 获取状态描述
 */
export function getStateDesc(state) {
  return ReportStateDesc[state] || '未知'
}

/**
 * 获取状态标签类型
 */
export function getStateType(state) {
  return ReportStateType[state] || 'info'
}

/**
 * 获取触发器描述
 */
export function getTriggerDesc(trigger) {
  return ReportTriggerDesc[trigger] || '未知'
}

File: ruoyi-system/src/main/java/com/ruoyi/system/domain/enums/ReportState.java
package com.ruoyi.system.domain.enums;

/**
 * 实验报告状态枚举
 */
public enum ReportState {
    /**
     * 草稿（未提交）
     */
    DRAFT("0", "草稿"),

    /**
     * 已提交（待批阅）
     */
    SUBMITTED("1", "已提交"),

    /**
     * 批阅中
     */
    REVIEWING("2", "批阅中"),

    /**
     * 已批阅（通过）
     */
    REVIEWED("3", "已批阅"),

    /**
     * 打回（需修改）
     */
    REJECTED("4", "已打回"),

    /**
     * 重新提交
     */
    RESUBMITTED("5", "重新提交"),

    /**
     * 已归档
     */
    ARCHIVED("6", "已归档");

    private final String code;
    private final String description;

    ReportState(String code, String description) {
        this.code = code;
        this.description = description;
    }

    public String getCode() {
        return code;
    }

    public String getDescription() {
        return description;
    }

    /**
     * 根据代码获取状态
     */
    public static ReportState fromCode(String code) {
        for (ReportState state : values()) {
            if (state.code.equals(code)) {
                return state;
            }
        }
        throw new IllegalArgumentException("未知的状态代码: " + code);
    }
}
